src/EventListener/ReadListener.php

```diff
@@ -101,10 +110,17 @@ private function getCollectionData(Request $request, array $attributes)
     private function getItemData(Request $request, array $attributes)
     {
         $id = $request->attributes->get('id');
+        $identifiers = [];
+        $context = [];
 
         try {
-            $data = $this->itemDataProvider->getItem($attributes['resource_class'], $id, $attributes['item_operation_name']);
-        } catch (PropertyNotFoundException $e) {
+            if ($this->identifierNormalizer) {
+                $identifiers = $this->identifierNormalizer->normalize((string) $id, $attributes['resource_class']);
+                $context = [ChainIdentifierNormalizer::HAS_IDENTIFIER_NORMALIZER => true];
+            }
+
+            $data = $this->itemDataProvider->getItem($attributes['resource_class'], $identifiers ?: $id, $attributes['item_operation_name'], $context);
+        } catch (InvalidIdentifierException $e) {
```

Comment:
@dunglas this is how I'd like things to be, although I think that this will break current DataProvider implementations. 
If we keep this in the data provider, implementing gets more complex and this path improves DX (end developper doesn't care about identifiers and gets what he expects, for example Uuid/Datetime instance instead of string).

Reply:
ping @dunglas thoughts about using `$context[ChainIdentifierNormalizer::HAS_IDENTIFIER_NORMALIZER]`?

Reply:
Looks good to me.